tests/rootfs.sh

```diff
@@ -0,0 +1,149 @@
+#!/bin/sh
+
+set -eu
+
+ERROR=1
+OK=0
+MY_NAME=$(basename -- "$0")
+ROOTFS_DIRECTORY="./rootfs"
+TEST_IMAGE_FILE_PATH="${ROOTFS_DIRECTORY}/tmp/test.img"
+TEST_EMPTY_MTAB_FILE="${ROOTFS_DIRECTORY}/etc/mtab"
+
+test_setup()
+{
+    if [ ! -f "${ROOTFS_DIRECTORY}/usr/bin/qemu-arm-static" ]
+    then
+        cp "$(command -v qemu-arm-static)" "${ROOTFS_DIRECTORY}/usr/bin/"
+    fi
+
+    dd if=/dev/zero of="${TEST_IMAGE_FILE_PATH}" bs=2M count=64 >/dev/null 2>&1 || return ${ERROR}
+    touch "${TEST_EMPTY_MTAB_FILE}" >/dev/null 2>&1 || return ${ERROR}
+    mkfs.ext4 "${TEST_IMAGE_FILE_PATH}" >/dev/null 2>&1 || return ${ERROR}
+    mount -o bind /dev/ "${ROOTFS_DIRECTORY}/dev/" >/dev/null 2>&1 || return ${ERROR}
+}
+
+test_destroy()
+{
+    umount -l "${ROOTFS_DIRECTORY}/dev"
+
+    if [ -f "${TEST_IMAGE_FILE_PATH}" ]
+    then
+        rm -f ${TEST_IMAGE_FILE_PATH}
+    fi
+
+    if [ -f "${TEST_EMPTY_MTAB_FILE}" ]
+    then
+        rm -f "${TEST_EMPTY_MTAB_FILE}"
+    fi
+    if [ -f "${ROOTFS_DIRECTORY}/usr/bin/qemu-arm-static" ]
+    then
+        rm -f "${ROOTFS_DIRECTORY}/usr/bin/qemu-arm-static"
+    fi
+}
+
+run_test()
+{
+    if ! test_setup
+    then
+        printf "Cannot run tests, unable to complete test setup\\n"
+        test_destroy
+        exit ${ERROR}
+    fi
+
+    if "$1"
+    then
+        printf "Run: %s - OK\\n" "${1}"
+    else
+        printf "Run: %s - ERROR\\n" "${1}"
+    fi
+    test_destroy
+}
+
+test_execute_resize2fs()
+{
+    ( chroot "${ROOTFS_DIRECTORY}" /sbin/resize2fs /tmp/test.img >/dev/null 2>&1 && return "${OK}" ) || return "${ERROR}"
+}
+
+test_execute_fdisk()
+{
+    ( chroot "${ROOTFS_DIRECTORY}" /sbin/fdisk -v >/dev/null 2>&1 && return "${OK}" ) || return "${ERROR}"
```

Comment:
can we do "$(command -v fdisk)" in the subshell instead?
and then fdisk --version (without silencing it) so we know a) it worked and b) which version (looking back at console logs this can be very valuable)
the long opt so it's immediately clear to the reader here, and not silenced so we can see errors and understand why!

Reply:
command is not part of the rootfs, so this can be used. Anyhow -v tests execution anyway so it is not needed. 
I agree with not silencing stderr, but not for stdout.